### PR TITLE
Extend SearchBeatmapSetsRequest by adding page parameter

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

File: osu.Game/Online/API/Requests/SearchBeatmapSetsRequest.cs [hunk 1]
```diff
@@ -11,23 +12,38 @@ namespace osu.Game.Online.API.Requests
...
         {
             this.query = System.Uri.EscapeDataString(query);
             this.ruleset = ruleset;
+            this.page = page;
             this.searchCategory = searchCategory;
             this.sortCriteria = sortCriteria;
             this.direction = direction;
         }
 
         // ReSharper disable once ImpureMethodCallOnReadonlyValueField
```
Comment (jorolf) [inline osu.Game/Online/API/Requests/SearchBeatmapSetsRequest.cs]: this can probably be removed now

File: osu.Game/Overlays/DirectOverlay.cs [hunk 2]
```diff
@@ -213,11 +216,12 @@ private void recreatePanels(PanelDisplayStyle displayStyle)
                 })
             };
 
+            loadCancellation = new CancellationTokenSource();
+
             LoadComponentAsync(newPanels, p =>
             {
-                if (panels != null) ScrollFlow.Remove(panels);
```
Comment (peppy) [inline osu.Game/Overlays/DirectOverlay.cs]: why was this removed?

**Comment by peppy** [comment]
Is this PR supposed to add pagination? The branch name suggests it is but it sin't.

**Comment by EVAST9919** [comment]
It was, but I thought it would be better to extend the request in it's own pr first

**Comment by peppy** [comment]
Need some way to test, though.

**Comment by peppy** [comment]
This seems weird now. You're adding a bindable which is going to be removed (or never used) in future changes.

If you can't test a change without adding cruft, you need to include more of what you changed. Or less. I would argue what you're doing here is pointless to PR separately from the actual pagination implementation.

**Comment by EVAST9919** [comment]
How we want to handle the pagination? Like infinite scroll or by just adding a ShowMore button?

**Comment by EVAST9919** [comment]
Will be reopened when after the actual scrolling implementation
